src/data/proxy/UserProfileStore.java

```diff
@@ -1,44 +1,28 @@
 package data.proxy;
 
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.function.Predicate;
 
-import data.proxy.utils.FunctionalUtils;
 import data.structure.UserProfile;
 
 /**
  * UserProfileStore manages access to the stored user profiles.
  */
-public class UserProfileStore {
-    private Map<String, UserProfile> userProfiles;
-    
-    /**
-     * Basic default constructor for UserProfileStore.
-     */
-    public UserProfileStore() {
-        userProfiles = new HashMap<String, UserProfile>();
-    }
+public abstract interface UserProfileStore {
```

Comment:
I generally like including something in the name that readily identifies this as an interface, perhaps IUserProfileStore. Thoughts?